bindata/manifests/machine-config/files/configure-switchdev.sh.yaml

```diff
@@ -6,18 +6,41 @@ contents:
     #!/bin/bash
     set -eux
     input="/etc/switchdev.conf"
+    UDEV_RULE_FILE='/etc/udev/rules.d/10-persistent-net.rules'
+
     if [ ! -f $input ]; then
       echo "File /etc/switchdev.conf not exist."
       exit
     fi
 
+    append_to_file(){
+      if ! test -f "$2"
+      then
+        echo "$1" > "$2"
+      else
+        if ! grep -Fxq "$1" "$2"
+        then
+          echo "$1" >> "$2"
+        fi
+      fi
+    }
+
+    add_udev_rule_for_sriov_pf(){
+        pf_pci=$(grep PCI_SLOT_NAME /sys/class/net/$1/device/uevent | cut -d'=' -f2)
+        udev_data_line="SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", KERNELS==\"$pf_pci\", NAME=\"$1\""
+        append_to_file "$udev_data_line" "$UDEV_RULE_FILE"
+    }
+
     names=()
 
     while read pci_addr num_vfs
     do
       echo "Set $num_vfs VFs on device $pci_addr"
 
       names+=($(ls /sys/bus/pci/devices/${pci_addr}/net/))
+      # Create udev rule to save PF name
+      add_udev_rule_for_sriov_pf $pci_addr
```

Comment:
Should the argument of add_udev_rule_for_sriov_pf being pf name?

Reply:
You r right I missed it before pushing it, it should be like
```
 cat /etc/udev/rules.d/10-persistent-net.rules 
SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", KERNELS=="0000:03:00.1", NAME="enp3s0f1np1"
```